packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -238,10 +257,54 @@ export class Summarizer extends EventEmitter implements ISummarizer {
     }
 
     public readonly summarizeOnDemand: ISummarizer["summarizeOnDemand"] = (...args) => {
-        if (this._disposed || this.runningSummarizer === undefined || this.runningSummarizer.disposed) {
-            throw Error("Summarizer is not running or already disposed.");
+        try {
+            if (this._disposed) {
+                throw Error("Summarizer is already disposed.");
```

Comment:
UsageError? Same question for other places where we through errors that might be usage errors